[RA-17] 알람 Add 화면 Present 방식 변경

jiyeonO opened the pull request:
- 상황: 기존에 알람 Modify 시 Sheet로 작업, 하지만 다시울림 & 직접 설정 & 알림음 설정 등 Sheet위에 Sheet로 올라가는 경우 다수 발생
- 수정 내용: Add 화면을 FullScreen으로 띄우도록 변경
- [Jira](https://davidyoon1122.atlassian.net/browse/RA-17?atlOrigin=eyJpIjoiNmZlMWI4NjAzODU4NDZlZDg1ZDRjZGM4MGI4ZGRhOTUiLCJwIjoiaiJ9)

## Add 화면 위에 Sheet 방식으로 올라가는 화면
<img width="619" alt="스크린샷 2024-12-04 오후 3 34 09" src="https://github.com/user-attachments/assets/491200a5-7bed-43f9-85ae-da5ce7e80a25">

Commit:
[RA-17] 알람 Add 화면 Present 방식 변경
- 상황: 기존에 알람 Modify 시 Sheet로 작업, 하지만 다시울림 & 직접 설정 & 알림음 설정 등 Sheet위에 Sheet로 올라가는 경우 다수 발생
- 수정 내용: Add 화면을 FullScreen으로 띄우도록 변경

davidyoon891122 commented on IntervalAlarm/Feature/Main/MainFeature.swift:
```diff
@@ -145,18 +144,18 @@ struct MainFeature {
+                                  .send(.sendNotification(alarm)))
+                case .element(id: _, action: .modified(.setAlarmOn(let alarm))):
+                    return .merge(.send(.onAppear),
+                                  .send(.removeNotification(alarm)),
```
이친구들은 순서 상관 없을까요 ?

jiyeonO replied:
리뷰 반영 f5b2545827a088d8a248b27585f006f5bf3e57cf

davidyoon891122 commented on IntervalAlarm/Feature/Main/MainFeature.swift:
```diff
@@ -235,16 +231,13 @@ struct MainView: View {
+                    case let .add(store):
+                        AddAlarmView(store: store)
                     case let .modified(store):
                         AddAlarmView(store: store)
```
(P2)
동일한 뷰에 동일한 스토어가 들어가는데 구분할 이유가 있나요 ?

jiyeonO replied:
`AddAlarmView`는 1) 신규 알람 추가 2) 기존 알람 변경에서 호출이 되는데요
- 위 두 가지의 경우에 따라 알람 저장하고 나서 처리가 달라서 구분이 필요했어요

> 어제의 대화 & 결론 :
> 같은 화면으로 보내니 Path 구분 하지 말고
> EntryType과 Action을 분기해서 처리하자

jiyeonO replied:
리뷰 반영 b8d6489703784b31b97c31f64444d0f0d5e1ee2c

jiyeonO committed:
[BugFix][RA-4] 알람 수정 시 Crash 발생 & UI 미반영 현상 (#22)

* [BugFix][RA-4] 알람 수정 시 Crash 발생 & UI 미반영 현상
- 발생 현상 : 알람 수정 시 Crash 발생
- 발생 원인 : 이미 존재하는 알람을 중복으로 추가 후 화면에서 그릴 때 id가 겹쳐서 발생하는 문제
- 수정 방법 : 이미 존재하는 알람을 수정하는 경우 Userdefault에서 교체

- 발생 현상 : 요일 선택 시 메인 UI에 미반영 되는 현상
- 발생 원인 : WeekdayDisplayView.weekdaySymbols에서 locale 미처리로 인해 같은 값으로 인식하지 못하는 문제
- 수정 방법 : AlarmModel.weekdaySymbols와 동일한 상태로 변경 처리

* 리뷰 반영
- weekdaySymbols을 전역변수로 Constants에 선언

---------

Co-authored-by: jiyeonO <jiyeonO>

Commit:
리뷰 반영
- 상황 : AddAlarmView는 1) 신규 알람 추가 2) 기존 알람 변경 두가지 경우에 호출
- 결론 : 같은 화면 호출이니 두 케이스를 Path로 구분하지 말고 Action으로 구분하도록 하자

Commit:
리뷰 반영
- 순서 유지 필요한 이슈 발견 수정

Commit:
Merge branch 'develop' into feature/RA-17

# Conflicts:
#	IntervalAlarm.xcodeproj/project.pbxproj
#	IntervalAlarm/Feature/Add/AddAlarmFeature.swift
